aiida/cmdline/commands/code.py

```diff
@@ -1039,13 +1015,73 @@ def code_delete(self, *args):
         print "Code '{}' deleted.".format(pk)
 
 
+def is_on_computer(ctx):
+    return bool(ctx.params.get('on_computer'))
+
+
+def is_not_on_computer(ctx):
+    return bool(not is_on_computer(ctx))
+
+
+def ensure_scripts(pre, post, summary):
+    if (not pre) or (not post):
+        return edit_pre_post(pre, post, summary)
+    return pre, post
+
+
+REMOTE_ABS_PATH = click.option(
+    '--remote-abs-path', prompt='Remote path', required_fn=is_on_computer, type=click.Path(file_okay=True), cls=InteractiveOption, help=('[if --installed]: The (full) absolute path on the remote ' 'machine'))
+
+
+@code_cmd.command('setup')
+@click.pass_context
+@options.LABEL(prompt='Label', cls=InteractiveOption)
+@options.DESCRIPTION(prompt='Description', cls=InteractiveOption)
+@click.option('--on-computer/--store-upload', is_eager=False, default=True, prompt='Installed on remote Computer?', cls=InteractiveOption)
+@options.INPUT_PLUGIN(prompt='Default input plugin', cls=InteractiveOption)
+@options.COMPUTER(prompt='Remote Computer', cls=InteractiveOption, required_fn=is_on_computer)
+@REMOTE_ABS_PATH
+@click.option('--code-folder', prompt='Folder containing the code', type=click.Path(file_okay=False, exists=True, readable=True), required_fn=is_not_on_computer, cls=InteractiveOption, help=('[if --upload]: folder containing the executable and ' 'all other files necessary for execution of the code'))
+@click.option('--code-rel-path', prompt='Relative path of the executable', type=click.Path(dir_okay=False), required_fn=is_not_on_computer, cls=InteractiveOption, help=('[if --upload]: The relative path of the executable file inside ' 'the folder entered in the previous step or in --code-folder'))
+@options.PREPEND_TEXT()
+@options.APPEND_TEXT()
+@options.NON_INTERACTIVE()
+@with_dbenv()
+def setup_code(ctx, non_interactive, **kwargs):
+    """Add a Code."""
+    from aiida.common.exceptions import ValidationError
+
+    if not non_interactive:
+        pre, post = ensure_scripts(kwargs.pop('prepend_text', ''), kwargs.pop('append_text', ''), kwargs)
+        kwargs['prepend_text'] = pre
+        kwargs['append_text'] = post
+
+    if kwargs.pop('on_computer'):
+        kwargs['code_type'] = CodeBuilder.CodeType.ON_COMPUTER
+    else:
+        kwargs['code_type'] = CodeBuilder.CodeType.STORE_AND_UPLOAD
+    code_builder = CodeBuilder(**kwargs)
+    code = code_builder.new()
+
+    try:
+        code.store()
+        code._reveal()  # newly setup code shall not be hidden
+    except ValidationError as err:
+        click.secho('Unable to store the code: {}. Exiting...'.format(err), fg='red', err=True)
+        raise click.Abort()
+
+    click.secho('Code "{}" successfully stored in DB.'.format(code.label), fg='green')
```

Comment:
Suggestion here: I created `aiida.cmdline.utils.echo` with some utility wrapper functions to use `click.secho` for some standard messages, such as success, error and warning. Would maybe also help uniformity if we use those where possible. In my other branch I am working on, I added `critical` which will also call `click.Abort()` but I will merge that later

Reply:
Also, where did you intend us to use `bold=True`?.

Reply:
Yes, I had already fixed the stderr omission. About the bold, I just put it there because sometimes I guess you want to really stress an error message potentially